wallet/wallet.c

```diff
@@ -31,6 +31,31 @@
 /* 12 hours is usually enough reservation time */
 #define RESERVATION_INC (6 * 12)
 
+/* Possible channel state */
+enum channel_state_db {
+	IN_OFFERED,
+	IN_FULLFILLED,
+	OUT_OFFERED,
+	OUT_FULLFILLED,
+};
+
+/* channel state identifier */
+struct channel_state_param {
+	const char *dir_key;
+	const char *type_key;
+	const enum channel_state_db state;
+};
+
+/* possible channels states */
+struct channel_state_param channel_state_params[] = {
+	{"in", "offered", IN_OFFERED}, {"in", "fulfilled", IN_FULLFILLED},
+	{"out", "offered", OUT_OFFERED}, {"out", "fulfilled", OUT_FULLFILLED},
+};
+
```

Comment:
If you decide to go with the bit trick below you can remove this struct array, since we'll just need 2 if-statements to cover all cases.

Reply:
The solution with enum is much better I think! Thanks